src/app/shared/models/attribute.ts

```diff
@@ -0,0 +1,3 @@
+export interface Attributes {
+  [key: string]: string
+}
```

Comment:
Why not use the built-in Map<String, String>? Benefits with all the built-in functions of the class.